### PR TITLE
File spec

### PR DESCRIPTION
**Pull request by bobthemighty**
With this request we introduce a new plugin for identifying contexts.

Use `run-contexts --specs=[PATH_TO_SPEC_FILE]` to load a list of files and directories that should be considered when running tests.

This is particularly useful if you have tests under multiple folders since you can call run-contexts once and have it fail the build.

The specs file should contain paths, one per line. Relative paths will be resolved from the current working directory when run-contexts is executed.

### CODE REVIEW

File: src/contexts/plugins/identification/filespec.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            action='store',
+            dest='specs',
+            default=None,
+            help="Path to a file containing files and directories to search for tests.")
+
+    def initialise(self, args=None, env=None, file=None, cwd=None):
+        """
+        Filthy hack: we provide file and cwd here rather than as constructor
+        args because Mr Hodgson decided to pass stdout as the only parameter
+        to __init__.
```
Comment (benjamin-hodgson) [inline src/contexts/plugins/identification/filespec.py]: Probably a mistake on my part. I've raised #7 to be fixed at a later date

**Comment by benjamin-hodgson** [comment]
The Travis build is failing because `pathlib` isn't available on Python 3.3. I'm afraid you'll have to restrict yourself to [`os.path`](https://docs.python.org/3.4/library/os.path.html) (comparing [`realpath`s](https://docs.python.org/3.4/library/os.path.html#os.path.realpath) will probably do what you want) for cross-compatibility.

**Comment by bobthemighty** [comment]
Ja, I noticed that. I'll try and fix it up later tonight. It's easy enough, just a bit messier :/

**Comment by benjamin-hodgson** [comment]
@bobthemighty is this ready to go?

**Comment by bobthemighty** [comment]
Ja, I think so, but I've not tested in depth. If you get bug reports/feature requests, fling them my way. Consider it working but probably underpowered.

**Pull request merged**